core/objects.go

```diff
@@ -265,16 +309,28 @@ func (ch Challenge) RecordsSane() bool {
 			ch.ValidationRecord[0].AddressUsed == nil || len(ch.ValidationRecord[0].AddressesResolved) == 0 {
 			return false
 		}
-	case ChallengeTypeDNS:
+	case ChallengeTypeDNS01:
 		// Nothing for now
 	}
 
 	return true
 }
 
-// IsSane checks the sanity of a challenge object before issued to the client
-// (completed = false) and before validation (completed = true).
-func (ch Challenge) IsSane(completed bool) bool {
+//----- BEGIN TO DELETE -----
+// IsLegacy returns true if the challenge is of a legacy type (i.e., one defined
+// before draft-ietf-acme-acme-00)
+func (ch Challenge) IsLegacy() bool {
+	return (ch.Type == ChallengeTypeSimpleHTTP) ||
+		(ch.Type == ChallengeTypeDVSNI)
+}
+
+// LegacyIsSane performs sanity checks for legacy challenge types, which have
+// a different structure / logic than current challenges.
+func (ch Challenge) LegacyIsSane(completed bool) bool {
```

Comment:
this doesn't need to be public. when it goes private, I'm not sure the isLegacy check below is as necessary.